parser-typechecker/src/Unison/Codebase/Causal.hs

```diff
@@ -69,6 +71,26 @@ data Causal m h e
           , tails :: Map (RawHash h) (m (Causal m h e))
           }
 
+toGraph
+  :: Monad m
+  => Set (RawHash h)
+  -> Causal m h e
+  -> m (Seq (RawHash h, RawHash h))
+toGraph seen c = case c of
+  One _ _           -> pure Seq.empty
+  Cons h1 _ (h2, m) -> if Set.notMember h1 seen
+    then do
+      tail <- m
+      g    <- toGraph (Set.insert h1 seen) tail
+      pure $ (h1, h2) Seq.<| g
+    else pure Seq.empty
+  Merge h _ ts -> if Set.notMember h seen
+    then do
+      tails <- sequence $ Map.elems ts
+      gs    <- Seq.fromList <$> traverse (toGraph (Set.insert h seen)) tails
+      pure $ Seq.fromList ((h, ) <$> Set.toList (Map.keysSet ts)) <> join gs
+    else pure Seq.empty
+
```

Comment:
Is this still used anywhere, and if not, should we use it or delete it or neither?

Reply:
This is a handy function for debugging Causals. What's the best thing to do, comment it out?

Reply:
No; maybe just a note that says it's currently kept around in case of debugging Causals — or you could wire it up to a ucm command, like `debug.history`, which I guess is similar but different.